Product/Production/Common/CONNECTCoreLib/src/main/java/gov/hhs/fha/nhinc/callback/openSAML/HOKSAMLAssertionBuilder.java

```diff
@@ -204,64 +196,64 @@ protected Issuer createIssuer(final CallbackProperties properties) {
+    protected Subject createSubject(final CallbackProperties properties, final X509Certificate certificate,
+        final PublicKey publicKey) throws SAMLComponentBuilderException {
         String x509Name = properties.getUsername();
 
-        if (NullChecker.isNullish(x509Name) || !checkDistinguishedName(x509Name)) {
-            if (null != certificate && null != certificate.getSubjectDN()) {
-                x509Name = certificate.getSubjectDN().getName();
-            }
+        if ((NullChecker.isNullish(x509Name) || !checkDistinguishedName(x509Name)) && null != certificate
+            && null != certificate.getSubjectDN()) {
+
+            x509Name = certificate.getSubjectDN().getName();
+
         }
-        return createSubject(x509Name, certificate, publicKey);
+        return createSubject(x509Name, publicKey);
     }
 
     /**
+     * Checks Distinguished Name, method have to be static, since it's private and no Instance variables are used, as
+     * per Sonar.
      *
      * @param userName
      * @return boolean
      */
-    private boolean checkDistinguishedName(final String userName) {
-        boolean isValid = true;
+    private static boolean checkDistinguishedName(final String userName) {
+        Name name = null;
         try {
-            final Name name = new LdapName(userName);
+            name = new LdapName(userName);
         } catch (final Exception e) {
             LOG.warn("Not a Valid Distinguished Name, setting the value from Certificate: {}", e.getLocalizedMessage());
             LOG.trace("Exception checking DN: {}", e.getLocalizedMessage(), e);
-            isValid = false;
         }
-        return isValid;
+        return name == null ? false : true;
     }
 
     protected Subject createEvidenceSubject(final CallbackProperties properties, final X509Certificate certificate,
-            final PublicKey publicKey) throws SAMLComponentBuilderException {
+        final PublicKey publicKey) throws SAMLComponentBuilderException {
 
         final String evidenceSubject = properties.getEvidenceSubject();
         String x509Name;
         if (NullChecker.isNullish(evidenceSubject)) {
             String userName = properties.getUsername();
 
-            if (NullChecker.isNullish(userName) || !checkDistinguishedName(userName)) {
-                if (null != certificate && null != certificate.getSubjectDN()) {
-                    userName = certificate.getSubjectDN().getName();
-                }
+            if ((NullChecker.isNullish(userName) || !checkDistinguishedName(userName)) && null != certificate
+                && null != certificate.getSubjectDN()) {
+
+                userName = certificate.getSubjectDN().getName();
+
             }
             x509Name = userName;
         } else {
             x509Name = evidenceSubject;
         }
-        return createSubject(x509Name, certificate, publicKey);
+        return OpenSAML2ComponentBuilder.getInstance().createSubject(x509Name, publicKey);
```

Comment:
why aren't you calling the protected method anymore?  if you're not going to use that method, remove it.